### ISSUE TITLE
Enabled mods get disabled when they are updated.

### ISSUE THREAD

**Issue by PetMudstone** [Regalis11/Barotrauma#9241]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

When you update mods that are enabled via the settings menu, they get disabled for some reason. This is quite annoying.

### Reproduction steps

1. Have an enabled mod that needs updating.
2. Update via settings menu.
3. Mod will be disabled automatically.

### Bug prevalence

Happens every time I play

### Version

0.17.15.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/0f071e7bd2820e01f3fbd0af108e7029603c07ae

**Comment by Regalis11** [comment]
Should be fixed in https://github.com/Regalis11/Barotrauma-development/commit/116d96e225a742c78a4d3a41f67361e655450f38

I also noticed another issue with the previous changes: the mod got moved to the end of the list when updated. That should now be fixed too.

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.

**Comment by NilanthAnimosus** [comment]
In testing issue #5717 on version 0.18.12.0 I have found this behavior still appears to exist. Updating all mods caused one of three to disappear from both lists and perhaps have no files in the installed AppData directory for the disabled/disappearing mods, additionally updating one at a time worked fine unless you press the "Accept" button in the specific mod preview when clicking on the update button for it (Without backing out before or after it completes updating). The giant accept/cancel buttons are far more easy to click than the little back arrow at the top left.

When a mod disappears from both lists, refreshing the mod list or reopening it will not bring it back, only a game restart will.
It is likely I caused a mod to disappear on update by clicking update individually -> updating it, and then "Accepting" the page. 

Not clicking accept and clicking the smaller back out button at the top left updates the mod correctly.

Note: this was replicated by modifying the installed (In Appdata) mods filelist installtime's and mod versions.

**Comment by Regalis11** [comment]
I'm afraid I couldn't reproduce this. If you can repro this consistently, could you please provide exact step-by-step instructions on how to do that? What do you mean by the "Accept" button (do you mean "Apply?").

**Comment by Regalis11** [comment]
Nvm, found one way to get it to happen now:
1. Move an outdated mod to the enabled mods list, but _don't apply the changes yet_
2. Go to the mod's details
3. Update the mod
4. Apply the changes in this view, without going back to the mod list

Addressed in https://github.com/Regalis11/Barotrauma-development/commit/7d9f422ed556afe7270438cd0a1e72ec46e103fc by disabling the "apply" button everywhere else except the mod list view.

**Comment by NilanthAnimosus** [comment]
Tested against dev commit https://github.com/Regalis11/Barotrauma-development/commit/17a4e29619e49adc0a329be654296cf8f8ddc5d6

noticed you can still update a mod and then have it disable if you do the steps below:

1. Move an outdated mod to the enabled mods list, but don't hit apply
2. Click the individual mod update button (Which opens its details).
3. Update the mod individually
4. Find the apply changes lights back up once it is finished
5. Hit apply
6. Back out and find the mod has changed back to the disabled list.

**Comment by Regalis11** [comment]
Yet another shot at this https://github.com/Regalis11/Barotrauma-development/commit/da30c3e9725c45e20b8c3ac945cdb358ffe500e6

**Comment by Rokvach** [comment]
Tested using the above repro steps, works correctly now. Closing.
